### PR TITLE
LoRA in Text2ImagePipeline

### PR DESCRIPTION
**Pull request by slyalin**

### CODE REVIEW

File: samples/cpp/stable_diffusion/CMakeLists.txt [hunk 1]
```diff
@@ -26,3 +26,22 @@ install(TARGETS stable_diffusion
         RUNTIME DESTINATION samples_bin/
         COMPONENT samples_bin
         EXCLUDE_FROM_ALL)
+
+# create LoRA sample executable
+
+add_executable(lora_stable_diffusion
```
Comment (ilya-lavrenov) [inline samples/cpp/stable_diffusion/CMakeLists.txt:32]: typically, we have a folder per sample with its own readme,md file, which describes GenAI capabilities demonstrated in the sample

File: src/cpp/src/llm_pipeline.cpp [hunk 2]
```diff
@@ -94,12 +95,9 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
...
     {
         ov::Core core;
-        auto adapters_iter = plugin_config.find(ov::genai::adapters.name());
-        if (adapters_iter != plugin_config.end()) {
-            m_generation_config.adapters = adapters_iter->second.as<AdapterConfig>();
-            auto filtered_plugin_config = plugin_config;
-            filtered_plugin_config.erase(ov::genai::adapters.name());
-            core.set_property(device, filtered_plugin_config);
+        if(auto filtered_plugin_config = extract_adapters_from_properties(plugin_config, m_generation_config.adapters))
+        {
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:99]: please, fix indentation

File: src/cpp/src/text2image/models/autoencoder_kl.cpp [hunk 3]
```diff
@@ -43,7 +44,7 @@ AutoencoderKL::AutoencoderKL(const std::string& root_dir,
                 const std::string& device,
                 const ov::AnyMap& properties)
     : AutoencoderKL(root_dir) {
-    compile(device, properties);
+    compile(device, filter_out_adapters_from_properties(properties).value_or(properties));
```
Comment (ilya-lavrenov) [inline src/cpp/src/text2image/models/autoencoder_kl.cpp:47]: looks like `.value_or(properties)` should be hidden inside `filter_out_adapters_from_properties`

File: src/cpp/src/lora_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+void get_adapters_from_properties (const AnyMap& properties, AdapterConfig& adapter_config) {
+    auto adapters_iter = properties.find(AdaptersProperty::name());
+    if (adapters_iter != properties.end()) {
+        adapter_config = adapters_iter->second.as<AdapterConfig>();
+    }
+}
+
+std::optional<AnyMap> filter_out_adapters_from_properties (const AnyMap& properties) {
+    return ::extract_adapters_from_properties(properties);
```
Comment (ilya-lavrenov) [inline src/cpp/src/lora_helper.cpp:38]:
```suggestion
    return ::extract_adapters_from_properties(properties).value_or(properties);
```
and remove std::optional from return value

File: samples/cpp/stable_diffusion/CMakeLists.txt (hunk 1 above)
Comment (slyalin) [reply samples/cpp/stable_diffusion/CMakeLists.txt:32]: What is the recommendation for other files that are shared between these samples: `imwrite.hpp/cpp`? Should we move them to a shared directory or duplicate in each sample directory making samples fully independent?

We are going to have adapters supported for each type of pipeline. There are two options for how we can organize the samples:

 - Create a dedicated sample for each combination LoRA + pipeline with its own directory; README will explain only LoRA-specific things explicitly referring to the "main" sample for the pipeline without LoRA. `lora_greedy_causal_lm` takes this path but doesn't have README yet.
 - Add `lora.cpp` file in each relevant existing sample; add LoRA section to the main README file for the sample. Old SD samples demonstrated that approach, and this PR gives an example how it looks but still README is not ready.

I don't see much difference between them, let's align and pick one.

Comment (ilya-lavrenov) [reply samples/cpp/stable_diffusion/CMakeLists.txt:32]: I'm Ok to have a folder per use case:
- text generation / llm
- text to image
- whisper
- visual language 

with multiple samples within a folder.
Each folder will have a common readme with generic use-case description and small per-sample section (e.g. what sample demonstrates, commands to get models and run app)

**Review by ilya-lavrenov — APPROVED** [review]

File: .github/workflows/stable_diffusion_1_5_cpp.yml [hunk 5]
```diff
@@ -64,12 +64,21 @@ jobs:
...
+      - name: Download LoRA adapter
+        run: |
+          wget -O ./build/samples/cpp/text2image/soulcard.safetensors https://civitai.com/api/download/models/72591
+
+      - name: Run LoRA app
+        run: |
+          source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
+          ./build/samples/cpp/text2image/lora_stable_diffusion ./models/dreamlike-art-dreamlike-anime-1.0/FP16 "curly-haired unicorn in the forest, anime, line" soulcard.safetensors 0.7
+
+    stable_diffusion_1_5_cpp-windows:
```
Comment (eaidova) [inline .github/workflows/stable_diffusion_1_5_cpp.yml:81]:
```suggestion
  stable_diffusion_1_5_cpp-windows:
```

**Pull request merged**